frame/session/src/historical.rs

```diff
@@ -274,6 +275,9 @@ impl<T: Trait, D: AsRef<[u8]>> frame_support::traits::KeyOwnerProofSystem<(KeyTy
 	type IdentificationTuple = IdentificationTuple<T>;
 
 	fn prove(key: (KeyTypeId, D)) -> Option<Self::Proof> {
+		//@note must take a context containing the index as paramter
+		// rather than depending on the current one since this func might
+		// be called at a later point of time
```

Comment:
Yeah and the `ProvingTrie` should only be generated if it's equal to the current session